lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -2240,6 +2240,29 @@ static bool foldMuxOfUniformArrays(MuxOp op, PatternRewriter &rewriter) {
   return true;
 }
 
+/// If the mux condition is an operand to the op defining its true or false
+/// value, replace the condition with 1 or 0.
+static bool assumeMuxCondInOperand(Value muxCond, Value muxValue,
+                                   bool constCond, PatternRewriter &rewriter) {
+  if (!muxValue.hasOneUse())
+    return false;
+  auto *op = muxValue.getDefiningOp();
+  if (!op)
```

Comment:
I think we need more restriction here. I think it's illegal to replace operands if op is observable in a different way (XMR to seq.firreg, hw.wire). 

I think it would be sufficient to restrict to combinatorial operations.

Reply:
Oh yeah excellent point :+1: Thanks!